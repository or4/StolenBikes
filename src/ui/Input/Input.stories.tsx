import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';

import { Input } from './Input';

const stories = storiesOf('Input', module);
stories.addDecorator(withKnobs);

stories.add('Input', () => {
    return <Input value={getTextKnobs()} placeholder={getPlaceholderKnobs()} />;
});

function getTextKnobs() {
    return text('Text', '');
}

function getPlaceholderKnobs() {
    return text('Placeholder', 'Enter your first name');
}
