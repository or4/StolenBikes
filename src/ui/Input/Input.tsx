import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import { theme } from '../theme';

// styles defined globally in App.css
export const StyledInput = styled.input``;

interface Props {
    placeholder: string;
    value: string;
}

export class Input extends React.PureComponent<Props> {
    public render() {
        const { ...rest } = this.props;

        const Input = <StyledInput {...rest} />;

        return <ThemeProvider theme={theme}>{Input}</ThemeProvider>;
    }
}
