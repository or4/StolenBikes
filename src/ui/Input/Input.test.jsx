import React from 'react';
import { mount } from 'enzyme';

import { Input } from './Input';

describe('Input', () => {
    test('should render correctly', () => {
        const component = <Input placeholder={'Enter your first name'} />;

        const mounted = mount(component);
        expect(mounted).toMatchSnapshot();
        mounted.unmount();
    });

    test('should expected value', () => {
        const component = <Input placeholder={'Enter your first name'} value={'Dmitry'} onChange={() => {}} />;
        const mounted = mount(component);
        expect(mounted.find('input').props().value).toEqual('Dmitry');
    });
});
