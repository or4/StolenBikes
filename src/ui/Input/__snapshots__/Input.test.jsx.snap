// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<Input
  placeholder="Enter your first name"
>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "buttonBackground": "rgba(0, 0, 0, 0.9)",
        },
        "fontSizes": Array [
          12,
          14,
          16,
          18,
          24,
          32,
          36,
          72,
          96,
        ],
        "space": Array [
          0,
          3,
          5,
          8,
          10,
          15,
          25,
        ],
      }
    }
  >
    <styled.input
      placeholder="Enter your first name"
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "lastClassName": "bDWFJH",
              "rules": Array [],
            },
            "displayName": "styled.input",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "input",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        placeholder="Enter your first name"
      >
        <input
          className="sc-bdVaJa bDWFJH"
          placeholder="Enter your first name"
        />
      </StyledComponent>
    </styled.input>
  </ThemeProvider>
</Input>
`;
